api/registry.py

```diff
@@ -94,6 +98,131 @@ def registrations(self):
         for x in self.integration.libraries:
             yield Registration(self, x)
 
+    def fetch_catalog(self, catalog_url=None, do_get=HTTP.debuggable_get):
```

Comment:
This code was moved in from Registration, and it properly belongs here. The main catalog and the registration document are located on the remote service, so the code for accessing them should be in RemoteRegistry.

I moved it because we now need to get the registration document at two different times: once when we're about to  register the circulation manager with a discovery service, and once when we're about to display the registration _form_ and we need to see 1) if the server's up and 2) if there are any terms of service we need to display.